dlib/matrix/matrix_utilities.h

```diff
@@ -48,6 +48,20 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    /*!A remove_complex
+        This is a template that can be used to remove std::complex from the underlying type.
+
+        For example:
+            remove_complex<float>::type == float
+            remove_complex<std::complex<float> >::type == float
+    !*/
+    template <typename T>
+    struct remove_complex {typedef T type;};
+    template <typename T>
+    struct remove_complex<std::complex<T> > {typedef T type;};
```

Comment:
You should make an alias of this that's `remove_complex_t`, as is done with all the similar type traits in the standard library, then client code is much simpler.  In fact, calling it `real_t` would be nice.  Then `std::is_same<real_t<complex<float>> , float>::value == true`

Reply:
I'll try using ```real``` and ```real_t```. But i'm wondering if i'm gonna get name conflicts with other parts of dlib

Reply:
I also quite like ```remove_complex``` as it follows the same theme as ```remove_const```, ```remove_cv```, ```remove_reference```, ```remove_pointer```, etc in ```<type_traits>```

Reply:
There is a `real`, so you can't use that name.  But there is no `real_t`.  But sure, `remove_complex` is nice for the reason you suggested .  But it's still better to have a `remove_const_t` to reduce the client boilerplate.